core/src/main/java/com/flipkart/gap/usl/core/registry/DimensionRegistry.java

```diff
@@ -266,4 +263,32 @@ private DimensionEvent parse(EventMapping eventMapping, ObjectNode data) throws
             throw new IngestionEventMappingException("Event should be of type " + DimensionEvent.class);
         }
     }
+
+    private JsonNode getEntityId(EventMapping eventMapping, ObjectNode data) throws IngestionEventMappingException{
+        if (eventMapping.getEntityIdPaths() != null)
+            return getEntityIdFromXPath(data, eventMapping.getEntityIdPaths());
+        Optional.ofNullable(eventMapping.getPivot()).orElseThrow(() ->
+                new IngestionEventMappingException("EntityId is missing from Dimension Event {}" + data));
+        return eventMapping.getPivot().getType().equals("xPath") ? getEntityIdFromXPath(data, eventMapping.getPivot().getValue()) :
+                getEntityIdFromGroovy(eventMapping.getPivot().getExpression(), data);
+    }
+
+    private JsonNode getEntityIdFromGroovy(String script, ObjectNode data) {
+        GroovyShell shell = new GroovyShell();
+        shell.setProperty("data", data);
+        String entityId = shell.evaluate(script).toString();
+        ObjectMapper mapper = ObjectMapperFactory.getMapper();
+        data.set("entityIdFromGroovy", mapper.convertValue(entityId, JsonNode.class));
```

Comment:
instead of convert value type cast it to JsonNode

Reply:
Use GroovyClassLoader to avoid creation of classes everytime..it can cache and provide better performance

Reply:
Changes done for GroovyClassLoader

Reply:
Groovy Script will return a string and I can't directly typecast string to JSON node. It will throw this exception:
java.lang.ClassCastException: java.lang.String cannot be cast to com.fasterxml.jackson.databind.JsonNode

That's why I am using mapper for this